src/libraries/System.Net.HttpListener/src/System/Net/Windows/HttpListener.Windows.cs

```diff
@@ -1579,8 +1579,8 @@ private static void SendError(HttpListenerSession session, ulong requestId, Http
                 httpResponse.pReason = (sbyte*)pReason;
                 httpResponse.ReasonLength = (ushort)byteReason.Length;
 
-                byte[] byteContentLength = Encoding.Default.GetBytes("0");
-                fixed (byte* pContentLength = &byteContentLength[0])
+                ReadOnlySpan<byte> byteContentLength = "0"u8;
+                fixed (byte* pContentLength = byteContentLength)
```

Comment:
this whole block is a bit weird - it fixes a single-byte array.
Should not it be just

```csharp
byte b = (byte)'0';
byte* pContentLength= &b;
```

Reply:
Maybe. I didn't want to mess with it too much as our coverage of HttpListener is spotty.

Reply:
> this whole block is a bit weird - it fixes a single-byte array. Should not it be just
> 
> ```cs
> byte b = (byte)'0';
> byte* pContentLength= &b;
> ```

Do note that this would remove the null terminator.